CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/SyntaxRewriterFile.swift

```diff
@@ -187,21 +201,16 @@ let syntaxRewriterFile = SourceFileSyntax(leadingTrivia: copyrightHeader) {
       """
       /// Interpret `node` as a node of type `nodeType`, visit it, calling
       /// the `visit` to transform the node.
+      @inline(__always)
       private func visitImpl<NodeType: SyntaxProtocol>(
-        _ node: Syntax,
+        _ node: inout Syntax,
```

Comment:
Out of curiosity: Do you know why `inout` is faster here? Does it avoid reference counting?

Reply:
I actually haven't measured performance difference between `inout` or not. I just followed `SyntaxVisitor` pattern here.

Reply:
Also since we do `isKnownUniquelyReferenced`, we can't do like `let newNode = dispatchVisit(oldNode)` because `newNode` and `oldNode` might be the same. So we need: `node = dispatchVisit(node)` anyway. which is semantically what `inout` does.

Reply:
Ah, I see. Makes sense 👍🏽